src/ai/backend/agent/agent.py

```diff
@@ -1300,9 +1325,12 @@ async def _handle_clean_event(self, ev: ContainerLifecycleEvent) -> None:
                     ev.kernel_id in self.restarting_kernels,
                 )
             except Exception as e:
+                log.exception("unhandled exception while processing CLEAN event: {0}", repr(e))
                 if ev.done_future is not None:
                     ev.done_future.set_exception(e)
                 await self.produce_error_event()
+            else:
+                log.info("Kernel {0} cleaned", ev.kernel_id)
```

Comment:
We are including it like this this time, but next time, let's apply it as a stage concept and accumulate indicators for each stage.